src/components/CollectionCard/CollectionCard.tsx

```diff
@@ -15,16 +18,33 @@ export type TCollectionCard = {
   onViewOnScanClick(): void
 };
 
-export const CollectionCard: FC<TCollectionCard> = ({ collection, onManageTokensClick, onRemoveCollectionClick, onViewOnScanClick }) => {
+export const CollectionCard: FC<TCollectionCard> = ({
+  collection,
+  onManageSponsorshipClick,
+  onRemoveSponsorshipClick,
+  onManageTokensClick,
+  onRemoveCollectionClick,
+  onViewOnScanClick
+}) => {
+  // TODO: remove this after the API provides complete collection details (cover, sponsorship, etc)
+  const { api } = useApi();
+  const collectionApi = api?.collection;
+  const [collectionDetails, setCollectionDetails] = useState<NFTCollection | null>();
+  useEffect(() => {
+    (async () => {
+      setCollectionDetails(await collectionApi?.getCollection(collection.id));
+    })();
+  }, [collection, collectionApi]);
+
   return (
     <CollectionCardStyled>
       <PictureWrapper>
-        <Picture alt={collection?.id?.toString() || ''} src={collection.coverImageUrl} />
+        <Picture alt={collection?.id?.toString() || ''} src={collection.coverImageUrl || collectionDetails?.coverImageUrl} />
         <ActionsMenuWrapper>
           <Dropdown placement={'right'}
             dropdownRender={() => (<DropdownMenu>
-              {/* <DropdownMenuItem onClick={onManageSponsorshipClick}>Manage sponsorship</DropdownMenuItem> */}
-              {/* <DropdownMenuItem onClick={onRemoveSponsorshipClick}>Remove sponsorship</DropdownMenuItem> */}
+              {collectionDetails?.sponsorship?.unconfirmed && <DropdownMenuItem onClick={onManageSponsorshipClick}>Manage sponsorship</DropdownMenuItem>}
```

Comment:
collectionDetails?.sponsorship

make a variable out of it. Being used in multiple places